app/services/ProcedureCodeService.scala

```diff
@@ -19,33 +19,49 @@ package services
 import connectors.{CodeLinkConnector, CodeListConnector}
 import models.codes.{AdditionalProcedureCode, ProcedureCode}
 import models.DeclarationType.{CLEARANCE, DeclarationType}
+import play.api.Logging
 
 import java.util.Locale
 import javax.inject.Inject
 
-class ProcedureCodeService @Inject()(codeListConnector: CodeListConnector, codeLinkConnector: CodeLinkConnector) {
+class ProcedureCodeService @Inject()(codeListConnector: CodeListConnector, codeLinkConnector: CodeLinkConnector) extends Logging {
 
   def getProcedureCodesFor(journey: DeclarationType, isEidr: Boolean, locale: Locale): Seq[ProcedureCode] =
     journey match {
       case CLEARANCE if !isEidr => codeListConnector.getProcedureCodesForC21(locale)
       case _                    => codeListConnector.getProcedureCodes(locale)
     }
 
+  def getProcedureCodeFor(procedureCode: String, journey: DeclarationType, isEidr: Boolean, locale: Locale): Option[ProcedureCode] =
+    (journey match {
+      case CLEARANCE if !isEidr => codeListConnector.getProcedureCodesForC21(locale).filter(_.code == procedureCode)
```

Comment:
Why not using a Map for the lookup?

Reply:
Because we are dealing with a small list of values so I did not see the need for a Map

Reply:
My suggestion to use a `Map` instead of a `List` was to simplify `CodeListConnector` trait, which could have then only one signature. And using a `Map` would not harm. O(1) vs O(n) for lookup.

Reply:
Ok, fair enough. As it also removes the need for a separate `readCodesFromFile` method in the connector I'll make this change :+1:

Reply:
Indeed. That was what I meant :+1: